### ISSUE TITLE
[REQUEST] Add option for pings to Monitor

### ISSUE THREAD

**Issue by danmed** [glanceapp/glance#2]
the Monitor service (for me anyway) would benefit from an option to react to just a ping rather than an http 200 code..

Would it be possible to add just an option so the full URl can be added (so it can be clicked) but the result of the monitor is based on a ping result?

**Comment by svilenmarkov** [comment]
Hey, thanks for providing feedback! 

One issue with pinging is that it typically requires root privileges which is something I wouldn't want to involve for an application that simply aggregates data from a bunch of HTTP requests. This leaves me with having to call the OS native `ping` command and parse the result, however since I want to maintain cross OS compatibility I have to potentially parse 3 different outputs. 

I'm not entirely sure what the best way to go about this is for the time being, I'll have to give it some thought.

**Comment by danmed** [comment]
Appreciate the reply and your consideration... maybe a way to customise the response code that means it's all good? for example when i try and add my home assistant instance it fails (yet clicking the link actually shows it's all fine...)

**Comment by svilenmarkov** [comment]
In what way does it fail? Can you check what the error is by hovering over the red "ERROR" text? There should also be an error in your logs.

**Comment by danmed** [comment]
ill go back and set up the monitor again.. i removed it because i realised i just wanted bookmarks.. uptime-kuma is already monitoring things for me..

Ill get back to you

**Comment by Petbotson** [comment]
hey,
i am encountering the same issue (or one like @danmed).

I have a few services listed, with a valid let's encrypt certificate, where everything is shown as fine.
But when it is a self signed certificate the check failes.

example one
`Get "domain.com": tls: failed to verify certificate: x509:  certificate relies on legacy Common Name field, use SANs instead`

example two
`Get "domain2.com": tls: failed to veryfy certificate: x509: certificate is valid for localhost, no domain2.com

I think it would be great to have some kind of toggle to accept insecure certificates.

**Comment by Bit0r** [comment]
Services like MySQL cannot accept HTTP requests and indeed require using ping to determine if the service is alive.

**Comment by Bit0r** [comment]
Backend services like MySQL cannot accept HTTP requests; their availability can only be confirmed through ping.

**Comment by Bit0r** [comment]
The Go language has a [pro-ping](https://github.com/prometheus-community/pro-bing) library, which can send ping using three different protocols: `icmp`, `udp`, and `http`.

**Comment by Rhys-Woolcott** [comment]
> hey, i am encountering the same issue (or one like @danmed).
> 
> I have a few services listed, with a valid let's encrypt certificate, where everything is shown as fine. But when it is a self signed certificate the check failes.
> 
> example one `Get "domain.com": tls: failed to verify certificate: x509: certificate relies on legacy Common Name field, use SANs instead`
> 
> example two `Get "domain2.com": tls: failed to veryfy certificate: x509: certificate is valid for localhost, not domain2.com`
> 
> I think it would be great to have some kind of toggle to accept insecure certificates.

Yeah, hard agree. Especially in environments where you have your own root authority.

**Comment by phil9309** [comment]
Is this issue still active? 

I'm a new user having the same problem for Plex and Wallos. Plex is online but the ping monitor shows red and "unauthorzied". So far I've been using a (gethomepage.dev) homepage. Their siteMonitor feature displays a green dot for HTTP Status 401 (Plex) or "too many redirects" (Wallos) too. Wouldn't that be the most simple solution to this?

**Comment by Petbotson** [comment]
hey @phil9309 
you can check out the [docs](https://github.com/glanceapp/glance/blob/main/docs/configuration.md#monitor)
You can set `allow-insecure`. Explanation: `Whether to ignore invalid/self-signed certificates`

**Comment by indivisionjoe** [comment]
Hello @svilenmarkov,

I have the same request, but for a different reason.

I’d like to monitor the availability of various devices on my network using ICMP. I previously did this with the Homepage dashboard, but I prefer the features of Glance.

The issue with using HTTP GET is that my network consists of multiple VLANs. For the GET requests to reach the server, I’d need to open firewall rules, which I’d rather avoid. ICMP feels like a more secure option than HTTP for device-to-device communication in my context.

**Comment by AlfaJackal** [comment]
Same here! Would love to have more options here because I also want to avoid opening my firewall just to send HTTP GET requests to different VLANs.

**Comment by SeimusS** [comment]
> Hello [@svilenmarkov](https://github.com/svilenmarkov),
> 
> I have the same request, but for a different reason.
> 
> I’d like to monitor the availability of various devices on my network using ICMP. I previously did this with the Homepage dashboard, but I prefer the features of Glance.
> 
> The issue with using HTTP GET is that my network consists of multiple VLANs. For the GET requests to reach the server, I’d need to open firewall rules, which I’d rather avoid. ICMP feels like a more secure option than HTTP for device-to-device communication in my context.

Same here. 

ICMP overall is used for reachability and doesn't introduce such potential security concern. Opening http/s or other ports the web service or GUI may run on, introduces a security concern as many times those are used for management of specific service/app itself.
